lib/Schedule/LongSteps.pm

```diff
@@ -455,4 +461,43 @@ sub find_process{
     return $self->storage()->find_process($pid);
 }
 
+sub load_process {
+    my ( $self, $pid, $context ) = @_;
+    $context ||= {};
+
+    my $loaded_process;
+    eval {
+        my $stored_process = $self->find_process($pid);
+        $loaded_process = $self->_load_process( $stored_process, $context );
+    };
+
+    if ($@) {
+        $log->critical("Failed to load the process: $@");
+    }
+
+    return $loaded_process;
+}
+
+
+# make sure you put this in an eval.
```

Comment:
again the why here is load class can croak, and because LS has error call backs, in the main code, the code calling it should deal with its errors and not this private method.